### ISSUE TITLE
Unit flake in k8s.io/kubernetes/contrib/mesos/pkg/queue

### ISSUE THREAD

**Issue by brendandburns** [kubernetes/kubernetes#26721]
From:
https://pantheon.corp.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/26389/kubernetes-pull-test-unit-integration/28888/

```
ok  	k8s.io/kubernetes/contrib/mesos/pkg/podutil	1.436s
ok  	k8s.io/kubernetes/contrib/mesos/pkg/proc	4.400s
FAIL	k8s.io/kubernetes/contrib/mesos/pkg/queue	3.608s
ok  	k8s.io/kubernetes/contrib/mesos/pkg/redirfd	1.076s
```

No additional info (not sure why?)

**Comment by ncdc** [comment]
In the future, would you mind posting public links? 😄 

From https://console.cloud.google.com/m/cloudstorage/b/kubernetes-jenkins/o/pr-logs/pull/26389/kubernetes-pull-test-unit-integration/28888/artifacts/junit_v1%2Cextensions-v1beta1%2Cfederation-v1alpha1_20160602-101104.stdout
```
--- FAIL: TestDQ_ordered_add_pop (3.00s)
	delay_test.go:211: totalDelay < 3s: 2.889987265s
```

**Comment by karlkfi** [comment]
@jdef is this solved by #24916 ?
